torch/_tensor_str.py

```
@@ -127,12 +128,33 @@ def tensor_totype(t):
 class _Formatter:
     def __init__(self, tensor):
         self.floating_dtype = tensor.dtype.is_floating_point
         self.int_mode = True
         self.sci_mode = False
         self.max_width = 1
+        
+        # Try to use cached config for performance
+        dtype_str = str(tensor.dtype)
+        device_str = str(tensor.device)
+        numel = tensor.numel()
+        cached_config = _get_tensor_format_config(dtype_str, device_str, numel)
```

Ruff (0.14.6)

Local variable `cached_config` is assigned to but never used

Remove assignment to unused variable `cached_config`

(F841)